homeassistant/components/modbus/climate.py

```diff
@@ -66,48 +68,69 @@
         vol.Optional(CONF_MAX_TEMP, default=35): cv.positive_int,
         vol.Optional(CONF_MIN_TEMP, default=5): cv.positive_int,
         vol.Optional(CONF_STEP, default=0.5): vol.Coerce(float),
+        vol.Optional(CONF_STRUCTURE, default=">f"): cv.string,
         vol.Optional(CONF_UNIT, default="C"): cv.string,
     }
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Modbus Thermostat Platform."""
-    name = config[CONF_NAME]
-    modbus_slave = config[CONF_SLAVE]
-    target_temp_register = config[CONF_TARGET_TEMP]
-    current_temp_register = config[CONF_CURRENT_TEMP]
-    current_temp_register_type = config[CONF_CURRENT_TEMP_REGISTER_TYPE]
-    data_type = config[CONF_DATA_TYPE]
-    count = config[CONF_COUNT]
-    precision = config[CONF_PRECISION]
-    scale = config[CONF_SCALE]
-    offset = config[CONF_OFFSET]
-    unit = config[CONF_UNIT]
-    max_temp = config[CONF_MAX_TEMP]
-    min_temp = config[CONF_MIN_TEMP]
-    temp_step = config[CONF_STEP]
+    structure = config[CONF_STRUCTURE]
     hub_name = config[CONF_HUB]
     hub = hass.data[MODBUS_DOMAIN][hub_name]
 
+    data_types = {
```

Comment:
Please create a constant for this dictionary and move it to the other constants at the top.